openverse_catalog/dags/common/storage/media.py

```diff
@@ -167,11 +168,16 @@ def _initialize_output_path(
                 "OUTPUT_DIR is not set in the environment. Output will go to /tmp."
             )
             output_dir = "/tmp"
-        if version is None:
-            version = CURRENT_VERSION[self.media_type]
-
-        datetime_string = datetime.now().strftime("%Y%m%d%H%M%S")
-        output_file = f"{provider}_{self.media_type}_v{version}_{datetime_string}.tsv"
+        version = f"v{version or CURRENT_VERSION[self.media_type]}"
+
+        path_components = [
+            provider,
+            self.media_type,
+            version,
+            datetime.now().strftime("%Y%m%d%H%M%S"),
+            tsv_suffix,
+        ]
+        output_file = ("_").join(filter(None, path_components)) + ".tsv"
```

Comment:
Hehe it's a little face 😁 `("_")` (as a real note though I think the parens can be removed):

```suggestion
        output_file = "_".join(filter(None, path_components)) + ".tsv"
```

Awesome use of `filter`!

Reply:
What does this `filter` do?